### PR TITLE
Eliminate Custody Logic for Phase 0

### PR DESCRIPTION
**Pull request by terenc3t**
This PR implements https://github.com/ethereum/eth2.0-specs/pull/488 and removes all the custody logic in phase 0. Accomplished the following:

- uint32 -> uint64, normalized everything to uint64 to align with the spec
- renamed casper slashing to attester slashing
- `SlashableVote` had a structure change, updated processing logic and tests to reflect that
- removed `data` in naming

### CODE REVIEW

File: proto/beacon/p2p/v1/types.proto [hunk 1]
```diff
@@ -212,7 +201,7 @@ message Deposit {
 
 message Exit {
   uint64 slot = 1;
-  uint32 validator_index = 2;
```
Comment (JustinDrake) [inline proto/beacon/p2p/v1/types.proto]: Interesting that you were using a `uint32`. I guess another reason to move to `uint64`.

Comment (terenc3t) [reply proto/beacon/p2p/v1/types.proto]: Hah yes. I'm moving everything to `uint64` :sweat

File: beacon-chain/core/blocks/block_operations.go [hunk 2]
```diff
@@ -246,25 +248,25 @@ func ProcessCasperSlashings(
 	return beaconState, nil
 }
 
-func verifyCasperSlashing(slashing *pb.CasperSlashing) error {
-	slashableVoteData1 := slashing.SlashableVoteData_1
-	slashableVoteData2 := slashing.SlashableVoteData_2
-	slashableVoteData1Attestation := slashableVoteData1.Data
-	slashableVoteData2Attestation := slashableVoteData2.Data
+func verifyAttesgerSlashing(slashing *pb.AttesterSlashing) error {
```
Comment (nisdas) [inline beacon-chain/core/blocks/block_operations.go]: `verifyAttesterSlashing`

File: beacon-chain/core/blocks/block_operations.go [hunk 3]
```diff
@@ -181,54 +183,54 @@ func verifyProposerSlashing(
...
+			"number of attester slashings (%d) exceeds allowed threshold of %d",
+			len(body.AttesterSlashings),
+			params.BeaconConfig().MaxAttesterSlashings,
 		)
 	}
-	for idx, slashing := range body.CasperSlashings {
-		if err := verifyCasperSlashing(slashing); err != nil {
-			return nil, fmt.Errorf("could not verify casper slashing #%d: %v", idx, err)
+	for idx, slashing := range body.AttesterSlashings {
+		if err := verifyAttesgerSlashing(slashing); err != nil {
```
Comment (nisdas) [inline beacon-chain/core/blocks/block_operations.go]: `verifyAttesterSlashing`

File: shared/slices/slice.go [hunk 4]
```diff
@@ -1,12 +1,12 @@
 package slices
 
-// Intersection of two uint32 slices with time
+// Intersection of two uint64 slices with time
 // complexity of approximately O(n) leveraging a map to
 // check for element existence off by a constant factor
 // of underlying map efficiency.
-func Intersection(a []uint32, b []uint32) []uint32 {
```
Comment (nisdas) [inline shared/slices/slice.go]: Can you remove the `uint64` specific slice operation functions below ?

File: beacon-chain/core/blocks/block_operations.go [hunk 5]
```diff
@@ -329,16 +323,26 @@ func casperSlashingPenalizedIndices(slashing *pb.CasperSlashing) ([]uint32, erro
...
+	if len(votes.ValidatorIndices) == 0 {
+		return errors.New("empty validator indices")
+	}
+	for i := 0; i < len(votes.ValidatorIndices)-1; i++ {
+		if votes.ValidatorIndices[i] >= votes.ValidatorIndices[i+1] {
+			return errors.New("validator indices not in descending order")
+		}
+	}
+	if len(votes.CustodyBitfield) != mathutil.CeilDiv8(len(votes.ValidatorIndices)) {
+		return errors.New("custody bit field don't match validator indices length")
```
Comment (nisdas) [inline beacon-chain/core/blocks/block_operations.go]: use formatting for the errors over here and below so that the error string is more descriptive

**Pull request merged**